inlang/source-code/cross-sell/cross-sell-sherlock/src/recommendation/recommendation.ts

```diff
@@ -0,0 +1,60 @@
+import type { NodeishFilesystem } from "@lix-js/fs"
+import { joinPath } from "./utils/joinPath.js"
+import { isExtensionsJson } from "./utils/isExtensionJson.js"
+import { parse, type CommentJSONValue, stringify } from "comment-json"
+import type { ExtensionsJson } from "./utils/types.js"
+
+export async function addRecommendationToWorkspace(
+	fs: NodeishFilesystem,
+	workingDirectory?: string
+): Promise<void> {
+	const vscodeFolderPath = joinPath(workingDirectory ?? "", ".vscode")
+	const extensionsJsonPath = joinPath(vscodeFolderPath, "extensions.json")
+
+	if (!(await fs.stat(vscodeFolderPath))) {
+		await fs.mkdir(vscodeFolderPath)
+	}
+
+	let extensions: ExtensionsJson
+	if (await fs.stat(extensionsJsonPath)) {
+		try {
+			const parsed = parse(await fs.readFile(extensionsJsonPath, { encoding: "utf-8" }))
+			if (isExtensionsJson(parsed)) {
+				extensions = parsed
+			} else {
+				extensions = { recommendations: [] }
+			}
+		} catch (error) {
+			extensions = { recommendations: [] }
+		}
+	} else {
+		extensions = { recommendations: [] }
+	}
+
+	if (!extensions.recommendations.includes("inlang.vs-code-extension")) {
+		extensions.recommendations.push("inlang.vs-code-extension")
+		await fs.writeFile(extensionsJsonPath, stringify(extensions, undefined, 2))
+	}
+}
+
+export async function isInWorkspaceRecommendation(
+	fs: NodeishFilesystem,
+	workingDirectory?: string
+): Promise<boolean> {
+	const vscodeFolderPath = joinPath(workingDirectory ?? "", ".vscode")
```

Comment:
Same as above

Reply:
done in https://github.com/opral/monorepo/pull/2360/commits/0995046e0633c70b5a85d440419e682950d3770c